### DISCUSSION TITLE
api /sp/targets/bid/recommendations v4 two questions?

### DISCUSSION THREAD

**Discussion by LionSniper** [amzn/ads-advanced-tools-docs#257] [Q&A] [answered]
@elizafarley hello, elizafarley

Question 1: i want to ask you when i request api /sp/targets/bid/recommendations v4, how to right set param "prime shipping eligible",  is it like this?  if i remove "prime-shipping-eligible", i will get bid answer, but when i add "prime-shipping-eligible", i always get a empty answer, so, is my usage of this parameter correct? thanks!

"targetingExpressions": [
        {
            "type": "PAT_CATEGORY_REFINEMENT",
            "value": "category=\"xxxxx\" brand=\"xxxx\" prime-shipping-eligible=TRUE"
        }
    ]

-------------------------------------------------------

Question 2: sometimes i request api /sp/targets/bid/recommendations v4, request body like: 
"targetingExpressions": [
        {
            "type": "PAT_ASIN",
            "value": "xxxxx"
        }
    ]

i will get response like: 
"bidRecommendationsForTargetingExpressions": [
                {
                    "targetingExpression": {
                        "type": "PAT_ASIN",
                        "value": "xxxxxx"
                    },
                    "bidValues": [
                        {
                            "suggestedBid": 0.02
                        },
                        {
                            "suggestedBid": 0.02
                        },
                        {
                            "suggestedBid": 0.02
                        }
                    ]
                }
            ]

but when i request like: 
"targetingExpressions": [
        {
            "type": "PAT_CATEGORY_REFINEMENT",
            "value": "category=\"xxxxxx\" price=1-3 rating=0-3 prime-shipping-eligible=TRUE"
        },
        {
            "type": "PAT_ASIN",
            "value": "xxxxxxx"
        }
    ]
i will get response like: 
"bidRecommendationsForTargetingExpressions":[]

so, Why does adding a new parameter “PAT_CATEGORY_REFINEMENT” affect the result of another parameter “PAT_ASIN”, I think that the results of parameters should be independent of each other, so why do they sometimes seem to influence each other? thanks!

**Comment by elizafarley** [comment, marked as answer]
For question 1, try this:

`{
 "type":"PAT_CATEGORY_REFINEMENT",
"value":"category=\"11961863031\" prime-shipping-eligible=\"true\""
}`

I'll look into question 2 and get back to you.
